topo/node/node.go

```diff
@@ -487,26 +487,31 @@ func (n *Impl) CreateService(ctx context.Context) error {
 		return nil
 	}
 	for k, v := range n.Proto.Services {
-		name := v.Name
-		if name == "" {
-			name = fmt.Sprintf("port-%d", k)
+		names := dedupServiceNames(append(v.Names, v.Name))
+		if names == nil {
```

Comment:
To be safe, this should be `if len(names) == 0` as a slice of zero length is not nil.

Reply:
Ack.